test/unit/elsa/registry_test.exs

```diff
@@ -0,0 +1,80 @@
+defmodule Elsa.RegistryTest do
+  use ExUnit.Case
+
+  @registry :elsa_registry
+
+  defmodule TestServer do
+    use GenServer
+
+    def start_link(args) do
+      name = Keyword.fetch!(args, :name)
+      GenServer.start_link(__MODULE__, args, name: name)
+    end
+
+    def init(args) do
+      {:ok, Map.new(args)}
+    end
+
+    def handle_info(msg, state) do
+      send(state.pid, msg)
+      {:noreply, state}
+    end
+  end
+
+  setup do
+    Process.flag(:trap_exit, true)
+    {:ok, pid} = Elsa.Registry.start_link(name: @registry)
+
+    on_exit(fn -> assert_down(pid) end)
+
+    :ok
+  end
+
+  test "registers and lookups pids by name" do
+    Agent.start_link(fn -> :agent_value end, name: {:via, Elsa.Registry, {@registry, :agent}})
+
+    assert :agent_value == Agent.get({:via, Elsa.Registry, {@registry, :agent}}, fn s -> s end)
+  end
+
+  test "pid is automatically removed when process exits" do
+    {:ok, pid} = Agent.start_link(fn -> :agent_value end, name: {:via, Elsa.Registry, {@registry, :agent}})
```

Comment:
what's the value of linking this to the test process and trapping exits for ExUnit vs. just `Agent.start`?

Reply:
I forgot Agent.start was a thing!